### PR TITLE
Add name property to StoreProduct

### PR DESCRIPTION
**Pull request by tonidero**
### Description
Currently `StoreProduct`, `GoogleStoreProduct`, `AmazonStoreProduct` and `TestStoreProduct` expose the `title` of the product. With Google, they have 2 fields for products:
- The [name](https://developer.android.com/reference/com/android/billingclient/api/ProductDetails#getName()) which actually has the product name
- The [title](https://developer.android.com/reference/com/android/billingclient/api/ProductDetails#getTitle()) which has the product name + app name

We were only exposing the `title`. We probably should also expose the `name`. This means that for `AmazonStoreProduct`, the title will be the same as the `name`, but I think that's fine as long as it's documented.

Additionally, this changes our paywalls to use the `name` property instead of the `title`, since it's more appropriate for the paywalls.

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ internal class VariableDataProvider(
     }
 
     fun productName(rcPackage: Package): String {
-        return rcPackage.product.title
+        return rcPackage.product.name
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:49]: This is a behavior change, so the product names don't include the app name.

File: purchases/src/test/java/com/revenuecat/purchases/amazon/helpers/dummies.kt [hunk 2]
```diff
@@ -55,8 +55,10 @@ fun stubStoreProductForAmazon(
         get() = type
     override val price: Price
         get() = price
+    override val name: String
+        get() = "An awesome name"
     override val title: String
-        get() = "An awesome title"
+        get() = "An awesome name (App name)"
```
Comment (NachoSoto) [inline purchases/src/test/java/com/revenuecat/purchases/amazon/helpers/dummies.kt:61]: 👌🏻

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt (hunk 1 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:49]: But also a bug fix :)

**Review by NachoSoto — APPROVED** [review]
Thanks for doing this!

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt (hunk 1 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:49]: Actually what do you think of putting this in a separate PR so we can cool out that we fixed the variable?

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:49]: Yeah, I can do that, will need to update a couple tests but that's ok, will just put them back in that PR 👍

Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt]: Separated in https://github.com/RevenueCat/purchases-android/pull/1569

**Pull request merged**